pkg/apis/triggers/v1alpha1/trigger_types.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2019 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// TriggerSpec represents a connection between TriggerBinding, Params,
+// and TriggerTemplate; TriggerBinding provides extracted values for
+// TriggerTemplate to then create resources from.
+type TriggerSpec struct {
+	Bindings []*TriggerBinding `json:"bindings"`
+	Template TriggerTemplate   `json:"template"`
+	// +optional
+	Name         string                `json:"name,omitempty"`
+	Interceptors []*TriggerInterceptor `json:"interceptors,omitempty"`
+	// ServiceAccount optionally associates credentials with each trigger;
+	// more granular authorization for
+	// who is allowed to utilize the associated pipeline
+	// vs. defaulting to whatever permissions are associated
+	// with the entire EventListener and associated sink facilitates
+	// multi-tenant model based scenarios
+	// TODO do we want to restrict this to the event listener namespace and just ask for the service account name here?
+	// +optional
+	ServiceAccount *corev1.ObjectReference `json:"serviceAccount,omitempty"`
+}
+
+// TriggerStatus defines the observed state of Trigger.
+type TriggerStatus struct {
+	// leave it blank for now
+}
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Trigger defines a mapping of an input event to parameters. This is used
+// to extract information from events to be passed to TriggerTemplates within a
+// Trigger.
+// +k8s:openapi-gen=true
+type Trigger struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// Spec holds the desired state of the Trigger
+	// +optional
+	Spec TriggerSpec `json:"spec"`
+	// +optional
+	Status TriggerStatus `json:"status,omitempty"`
```

Comment:
I think for now we don't require TriggerStatus or TriggerCRD controller.

Reply:
fair enough, let's remove this for now. We can always add it later!!